src/runtime_src/core/edge/drm/zocl/zocl_ctrl_ert.c

```diff
@@ -753,7 +762,15 @@ static int zert_probe(struct platform_device *pdev)
 	if (!zert->zce_ctrl_xgq_hdl)
 		zert_err(zert, "failed to initialize CTRL XGQ");
 
-	return 0;
+	zert_init_cus(zert);
+	zert->zce_max_cu_size = CTRL_XGQ_CU_SLOT_SIZE;
+	ret = zert_create_cu_xgqs(zert);
+	if (ret)
+		zert_err(zert, "failed to initialize CU XGQs");
+
+	zert->zce_config_completed = true;
```

Comment:
zce_config_completed is for mark the start and end of a configuration during xclbin download. since you've changed so that the CU is now configured and un-configured individually, do you still need this marker?

Reply:
Actually as per the existing code after end of the configuration we are updating the KDS. 
If we remove this part then after every CU/SCU configuration/un-configuration we need to update KDS. 
I am avoiding that many times KDS to update. Instead once the configuration is end then only I will update KDS.

Reply:
Why you can't update KDS every time a CU is configured or unconfigured? "KDS can't be updated many times" is a hidden assumption in the code logic which I'd like to avoid, unless it is a HW issue. The less assumptions like this in the code, the better. If you think that it is more complicated to make KDS accept update many times, you could do it in follow up PR. But, I'd like to understand the reason behind it.

Reply:
There is no issue of updating kds many times.  But as the the current implementation there is no way we can start the CUs/SCUs thread individually. We are actually looping MAX_CUS number of time for each KDS update. I just tried to avoid this.

Reply:
Got it. We can fix the configure/unconfigure CU now, then we can focus on KDS part of CU starting/stopping change later. Architecturally, CU / SCU should be able to start and stop individually since we are moving to multi-XCLBIN schema now. KDS shall not worry about each individual XCLBINs and only sees CU/SCUs.